### ISSUE TITLE
Python Source Can't be Opened After Edit + Invalid Path Displayed

### ISSUE THREAD

**Issue by cooperpellaton** [psychopy/psychopy#1761]
***TL;DR:** Open a valid psychopy experiment, open source, edit the source, save, close source. Reopen source from psychopy and the old-pre-edited source is there. Now you can't view the new source. ALSO, delete this newly edited source, open psycho py experiment, open source, view source. BUT this source no longer exists on the computer!*

Information:
1. *Version*: Produced on 1.84.2, but replicated on all other versions.

**In detail:** I started by opening a valid psychopy experiment my colleague had put together which had some bugs. I begin to debug by reading and editing the source. After making multiple modifications I save and close the source. I navigated to the location on disk of the source, and open it in a text editor to verify that these modifications were in fact saved. Now, I reopen the same experiment from earlier and run it, *but* low and behold, none of my edits have taken effect. Startled, I open the source in psychopy, and despite it saying it is pulling from the location on disk where I **know** that the correct, and edited version of the source file is, the one displayed is the old, prior version. 

Now, startled, I decide to delete the source. After removing it from disk I then reopen the psychopy experiment and click on edit source. Now, not only does psychopy let me view the source I just deleted but it also displays the path of the source being edited as the path of the source I just deleted (which now isn't on disk) **and** when I go to that location on disk the source isn't there.

I think what all of this is getting at is that there is some kind of indexing and connection between the `python` and `.psyexp` files going on under the covers, and now this is surfacing both by showing *invalid* and *incorrect* paths on disk in addition to making it difficult to propagate edits to the source of an experiment.

Below, I have provided: 
1. a gif of the problem, 
2. the python source for the edited code and 
3. the psychopy experiment which if opened will show a different source than the edit, though claiming it is at the same location on disk.

Files:
1. ![Available tomorrow.]()
2. [Source.](https://gist.github.com/cooperpellaton/429f5f4833a52fba444364e2d95a0455)
3. [Experiment.](https://gist.github.com/cooperpellaton/92e9cf5bed329be93101f920d333584f)

**Comment by peircej** [comment]
When you say 'source' I think you mean you're compiling the experiment into a Python script and editing the script. The Python is not the source of the experiment (the experiment file is actually an xml file.

Read the following and reopend this issue if something still seems not to be right afterwards:
https://discourse.psychopy.org/t/interaction-of-builder-and-coder-view/2367

cheers

**Comment by cooperpellaton** [comment]
This did answer my questions regarding the interaction between the experiment file and the source. However, the one remark about the invalid path still stands. After deleting the `.py` and then opening the experiment file and then the builder, it will show you a path of the `.py` which should exist but doesn't because it hasn't yet been generated by a compilation of the experiment. This is not an issue per say but it is incorrect to show that invalid path at that time as it can lead to confusion such as in my case.

**Comment by peircej** [comment]
I think I see what you mean. The step about deleting the python script is unnecessary which had me confused - if you simply compile the script from the Builder view it provides this as a default name but doesn't actually save the file until you press save. I guess we could hold off inserting the default name until save is pressed, but it seems nice to provide something more useful than "untitled.py"
